pkg/ast/spl/spl.peg

```diff
@@ -951,6 +956,301 @@ TcOptionCMD <- option:("usenull" / "useother" / "nullstr" / "otherstr") {
     return optionStr, nil
 }
 
+SpanLenOption <- number:(FloatAsString / IntegerAsString) timeScale:(TimeScale)? {
+    if timeScale != nil {
```

Comment:
nit: since this rule is specific to `bin` (it returns a `structs.BinSpanLength`), you should rename the rule to specify that (e.g., `BinSpanLenOption`)

Reply:
addressed in the latest commit.